pkg/common/cns-lib/volume/manager.go

```diff
@@ -565,27 +575,97 @@ func (m *defaultManager) DeleteVolume(ctx context.Context, volumeID string, dele
 			log.Errorf("ConnectCns failed with err: %+v", err)
 			return err
 		}
-		// Construct the CNS VolumeId list.
-		var cnsVolumeIDList []cnstypes.CnsVolumeId
-		cnsVolumeID := cnstypes.CnsVolumeId{
-			Id: volumeID,
+		var (
+			// Reference to the DeleteVolume task on CNS.
+			task *object.Task
+			// Boolean to determine whether to invoke CNS
+			// DeleteVolume for this request.
+			invokeCNS = true
+			// Name of the CnsVolumeOperationRequest instance.
+			instanceName = "delete-" + volumeID
+			// Local instance of DeleteVolume details that needs to
+			// be persisted.
+			volumeOperationDetails *cnsvolumeoperationrequest.VolumeOperationRequestDetails
+			// Boolen to determine if the idempotency feature is supported by
+			// the CSI version.
+			isIdempotencyHandlingEnabled = m.idempotencyHandlingEnabled && m.operationStore != nil
+		)
+		if isIdempotencyHandlingEnabled {
```

Comment:
I see 9 FSS checks introduced in this method. Maybe its better to have a separate DeleteVolume method with FSS enabled. It also makes the cleanup easy as we just need to get rid of the old method when FSS is permanently enabled.

Reply:
Refactored the logic in the latest comments

Reply:
Thanks for refactoring Delete and Create methods in Volume Manager. IMHO this looks way better than your first commit.